pkg/networkservice/common/heal/client_test.go

```diff
@@ -104,16 +106,17 @@ func TestHealClient_Request(t *testing.T) {
 			},
 		},
 	}
-
-	cond := func() bool {
-		select {
-		case <-onHealCh:
-			return true
-		default:
-			return false
-		}
+	ctx, cancel := context.WithTimeout(context.Background(), waitHealTimeout)
+	defer cancel()
+	select {
+	case <-ctx.Done():
+		require.FailNow(t, "timeout waiting for Heal event %v", time.Since(t1))
+		return
+	case <-onHealCh:
+		// All is fine, test is passed
+		break
 	}
-	require.Eventually(t, cond, waitForTimeout, tickTimeout)
+	logrus.Infof("passed with total time: %v", time.Since(t1))
```

Comment:
Interesting... this feels like it should be the same as require.Eventually... is there a known problem with require.Eventually?